shared/actions/chat/index.js

```diff
@@ -985,18 +990,34 @@ function* _changedActive(action: ChangedActive): SagaGenerator<any, any> {
 
 function* _badgeAppForChat(action: Constants.BadgeAppForChat): SagaGenerator<any, any> {
   const conversations = action.payload
-  const conversationUnreadCounts = conversations.reduce((map, conv) => {
-    const unreadCounts: Constants.UnreadCounts = {
-      total: conv.get('unreadMessages'),
-      badged: conv.get('badgeCounts')[`${isMobile ? CommonDeviceType.mobile : CommonDeviceType.desktop}`],
-    }
-    if (!unreadCounts.total) {
-      return map
-    } else {
-      return map.set(Constants.conversationIDToKey(conv.get('convID')), unreadCounts)
+  let totals: {[key: string]: number} = {}
+  let badges: {[key: string]: number} = {}
+
+  conversations.forEach(conv => {
+    const total = conv.get('unreadMessages')
+    if (total) {
+      const badged = conv.get('badgeCounts')[
+        `${isMobile ? CommonDeviceType.mobile : CommonDeviceType.desktop}`
+      ]
+      const conversationIDKey = Constants.conversationIDToKey(conv.get('convID'))
+      totals[conversationIDKey] = total
+      if (badged) {
+        badges[conversationIDKey] = badged
+      }
     }
-  }, Map())
-  yield put(Creators.updateConversationUnreadCounts(conversationUnreadCounts))
+  })
+
+  badges = I.Map(badges)
+  totals = I.Map(totals)
+
+  const oldBadge = yield select(s => s.entities.inboxUnreadCountBadge)
+  const oldTotal = yield select(s => s.entities.inboxUnreadCountTotal)
+  if (!I.is(oldBadge, badges)) {
```

Comment:
keep these as separate things so we don't re-render when one changes. Only update if the values actually change